core/model/message.go

```diff
@@ -79,3 +79,41 @@ type MessagesStats struct {
 	Read       *int64 `json:"read_count" bson:"read_count"`
 	Unread     *int64 `json:"not_read_count" bson:"not_read_count"`
 }
+
+///
+
+//InputMessage is passed by the adapters for creating a message in the core module
+type InputMessage struct {
+	OrgID string `json:"org_id"`
+	AppID string `json:"app_id"`
+
+	Priority int               `json:"priority"`
+	Subject  string            `json:"subject"`
+	Sender   *InputSender      `json:"sender"`
+	Body     string            `json:"body"`
+	Data     map[string]string `json:"data"`
+
+	//recipients related
+	Recipients             []InputMessageRecipient  `json:"recipients"`
+	RecipientsCriteriaList []InputRecipientCriteria `json:"recipients_criteria_list"`
+	Topic                  *string                  `json:"topic"`
+}
+
+// InputSender is passed by the adapters for creating a message in the core module
+type InputSender struct {
+	UserID string `json:"user_id"`
+	Name   string `json:"name"`
+}
+
+// InputMessageRecipient is passed by the adapters for creating a message in the core module
+type InputMessageRecipient struct {
```

Comment:
@petyos, I'm not sure if you care to worry about this now or not but I assume the docs should be updated as well with some of these changes

Reply:
Fixed! Thanks